docs/source-pytorch/versioning.rst

```diff
@@ -0,0 +1,67 @@
+
+API Stability
+*************
+
+In Lightning, all API and features are marked as either stable or experimental.
+
+Experimental API
+----------------
+
+Experimental APIs are labelled as experimental or beta in the documentation and/or in the release note and are considered unstable and should not be used in production.
+
+For experimental features, any of the following may be true.
+
+- The feature has unstable dependencies.
+- The API may change without notice in future versions.
+- The performance of the feature has not been verified.
+- The docs for this feature are under active development.
+
+Stable API
+----------
+
+Everything not specifically labelled as experimental is stable.
+
+For stable APIs, all of the following are true:
+- The API is not expected to change.
+- If anything does change, we show a deprecation warning before applying the breaking change following the rule described below.
+
+API Evolution
+*************
+
+Lightning's development is driven by research and best practices in a rapidly developing field of AI and machine learning. Change is inevitable and when it happens, the Lightning team is committed to minimizing user friction and maximizing ease of transition from one version to the next. We take backwards compatibility and reproducibility very seriously.
+
+For API removal, renaming or other forms of backwards-incompatible changes, the procedure is:
+
+#. A deprecation process is initiated at a minor version ``X``, producing a deprecation warning at runtime and in the documentation.
+#. The deprecated API remains unchanged during the deprecation phase.
+#. The deprecation warning lasts for two minor versions before the breaking change takes effect at ``X+2`` minor version.
+
+If your code raises no deprecation warning with the latest minor release, it will be compatible with the next minor release.
+
+The ``X+2`` rule is a recommendation and not a strict requirement. Shorter or longer deprecation cycles may apply to some cases. In the past, DDP2 was removed without a deprecation process because the feature was broken and unusable beyond fixing as discussed in `#12584 <https://github.com/Lightning-AI/lightning/issues/12584>`_. Also, `#10410 <https://github.com/Lightning-AI/lightning/issues/10410>`_ is an example where a longer deprecation applied. We deprecated the accelerator arguments, such as ``Trainer(gpus=...)``, in 1.7, however, because the APIs were so core that they would impact almost all use cases, we decided not to introduce the breaking change until 2.0.
+
+Python Support
+**************
+
+PyTorch Lightning follows `NEP 29 <https://numpy.org/neps/nep-0029-deprecation_policy.html>`_ which PyTorch also follows (`#74203 <https://github.com/pytorch/pytorch/issues/74203>`_).
+
+PyTorch Support
+***************
+
+PyTorch Lightning supports the latest four minor versions of PyTorch. For example, PyTorch Lightning 1.7 supports PyTorch 1.9, 1.10, 1.11 and 1.12.
```

Comment:
I've seen our PyTorch version support window mentioned in the past issues/PRs by some core members. Can we officially state the rule here?

Reply:
IMO yes, but I'm not sure if 1.8 is going to break this pattern. Can you clarify @Borda?

Reply:
I think we keep this window